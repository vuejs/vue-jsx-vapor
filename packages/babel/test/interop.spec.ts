import { transformSync } from '@babel/core'
import { describe, expect, test } from 'vitest'
import jsx from '../src/index'

describe('transform', () => {
  test('transform multiple components', () => {
    const { code } = transformSync(
      `const A = defineComponent(() => {
         defineVaporComponent(() => <div />)
         return () => <div />
       })
       const B = defineVaporComponent(() => {
        const C = defineComponent(() => <div />)
        const D = <div />
        return <div />
       })`,
      {
        filename: 'test.tsx',
        plugins: [[jsx, { interop: true }]],
      },
    )!
    expect(code).toMatchInlineSnapshot(`
      "import { template as _template } from 'vue';
      const _t00 = _template("<div></div>", true);
      const A = defineComponent(() => {
        defineVaporComponent(() => (() => {
          const n0 = _t00();
          return n0;
        })());
        return () => <div />;
      });
      const B = defineVaporComponent(() => {
        const C = defineComponent(() => <div />);
        const D = (() => {
          const n0 = _t00();
          return n0;
        })();
        return (() => {
          const n0 = _t00();
          return n0;
        })();
      });"
    `)
  })
})
