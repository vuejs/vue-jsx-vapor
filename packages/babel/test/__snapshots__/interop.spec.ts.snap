// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform > transform multiple components 1`] = `
"import { template as _template } from 'vue';
const t0 = _template("<div></div>", true),
  t1 = t0,
  t2 = t1;
const A = defineComponent(() => {
  defineVaporComponent(() => (() => {
    const n0 = t0();
    return n0;
  })());
  return () => <div />;
});
const B = defineVaporComponent(() => {
  const C = defineComponent(() => <div />);
  const D = (() => {
    const n0 = t1();
    return n0;
  })();
  return (() => {
    const n0 = t2();
    return n0;
  })();
});"
`;
